app/controllers/learning_materials_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class LearningMaterialsController < ApplicationController
+  skip_before_action :authenticate_user!, only: %i[index]
+
+  def index
+    @learningmaterials = LearningMaterial.all
```

Comment:
eager load `thumbnails` so that we don't end up with N+1 queries when you then check to see if it's `attached?` 

```suggestion
    @learning_materials = LearningMaterial.includes(:thumbnail).all
```

Reply:
I tried eagerloading using this approach ```@learning_materials = LearningMaterial.includes(:thumbnail).all``` and I got the following error:

![image](https://github.com/nairuby/arc_platform/assets/86318284/e029b601-4b5d-4543-a227-b6ad6d737319)

I investigated and found out the reason this will not work is because Active Storage does not create a direct association called ```:thumbnail```. 

I instead replaced ```:thumbnail``` with the suggested ```:thumbnail_blob``` which then worked as expected. I went a little further into the docs of Active Storage and found the ```with_attached_*``` method that is provided with Active Storage and I found that the following also worked perfectly fine. ```@learning_materials = LearningMaterial.with_attached_thumbnail.all```. Reference used [this](https://edgeapi.rubyonrails.org/classes/ActiveStorage/Attachment.html) [this](https://stackoverflow.com/questions/60557783/rails-how-to-eager-load-attachments-belonging-to-an-associated-model) and [this](https://api.rubyonrails.org/v5.2.4/classes/ActiveStorage/Attached/Macros.html#method-i-has_one_attached).

I settled with this option ```@learning_materials = LearningMaterial.includes(:thumbnail_blob).all``` because with ```includes```, it will allow for further filtering/ sorting of associated records.

This is well resolved :+1: